client/lib/media/actions.js

```diff
@@ -94,6 +94,12 @@ MediaActions.fetchNextPage = function( siteId ) {
 	if ( ! query.source ) {
 		wpcom.site( siteId ).mediaList( query, mediaReceived );
 	} else {
+		if ( 'pexels' === query.source ) {
+			// Pexels limits the number of API calls per month, and if we had the default
+			// limit of 20 media items, it takes at least 2 API calls to fill the first page
+			// with results. This saves us some of those API calls.
+			query.number = 40;
```

Comment:
I'm not keen on this special hardcoded limit, especially in such a low level position. Most APIs have limits - Google Photos does too - and Pexels have given us a very generous amount that I don't think we'll run out of soon.

If we do start having limit problems, or the WPCOM API gets abused which in turn causes rate limiting, then it should be tackled at a higher level in a consistent manner that applies across all services, with values stored in a central location

Reply:
That seems sensible, I'll get this removed.